indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/SeekableStreamSupervisorSpecTest.java

```diff
@@ -784,13 +781,72 @@ public void testSeekableStreamSupervisorSpecWithScaleOut() throws InterruptedExc
     int taskCountAfterScaleOut = supervisor.getIoConfig().getTaskCount();
     Assert.assertEquals(2, taskCountAfterScaleOut);
     Assert.assertTrue(
-        metricCapture
-            .getValues()
+        dynamicActionEmitter
```

Comment:
Instead of this whole logic, you can simply do:
```java
dynamicActionEmitter.verifyEmitted(
    SeekableStreamSupervisor.AUTOSCALER_REQUIRED_TASKS_METRIC,
    Collections.singletonMap(
        SeekableStreamSupervisor.AUTOSCALER_SKIP_REASON_DIMENSION,
        "minTriggerScaleActionFrequencyMillis not elapsed yet",
        1 // no of times you expect this metric to be emitted
    )
);
```

Reply:
Same suggestion in the other test.